### ISSUE TITLE
Bones theme crashes with Grav 1.6.6

### ISSUE THREAD

**Issue by ThosG** [getgrav/grav#2468]
After update to 1.6.6, Bones theme crashed with error

> "Property 'inlinejs_pipeline_before_excludes' does not exist in the object!"

Following advice from iusvar I added
`protected $inlinejs_pipeline_before_excludes;`
to the file 
`/system/src/Grav/Common/Assets.php`
on line 53 and was well.

Bones theme is at v 0.5 and hasn't been updated for 3 years afaics.

Over to you, folk.

**Comment by rhukster** [comment]
I'll take a look... cheers.

**Comment by rhukster** [comment]
Yup, i see.. i moved some things around, but only moved `css` and `js`, not the `inlinecss` and `inlinejs` vars.

I'll move those for next release.

**Comment by ThosG** [comment]
Thanks Andy.

I just hit another. After making the recommended patch I went to to http://www.greenmanpress-music.co.uk/arrangements/cvr-3208 <http://www.greenmanpress-music.co.uk/arrangements/cvr-3208> and clicked the link to Handel. Got 
Duplicate key "@page.descendants" detected at line 6 (near "'@page.descendants': '/largerworks'”).

It all worked fine with the previous version. 1.6.5, he said ruefully.

Should I rebuild the site with an up to date theme like Quark?

(Bit embarrassing, I’ve been paid for this site, but not the Grav team’s fault)

Thomas

**Comment by rhukster** [comment]
I would recommend using Quark for a base as that uses latest best practices.  We did go back and update all our themes to use the recommended asset structure, but didn't go any deeper than that.  This is one of the first themes we released, and as such many new features have been added, and better ways of doing things have been discovered.

However, your issue can probably be fixed.  The issue is that Grav 1.6 uses the latest and greatest libraries including Yaml parser which is more strict than it used to be.  Before it would allow broken or invalid yaml that it should not have, those issues are now appearing.  it's not because Grav 1.6 is broken, its actually because things were broken before, but you just didn't know it.

I would need to see the underlying .md files though to know what is causing the issue.  Can you zip it up and send it to me? perhaps via DM on Discord chat?

**Comment by ThosG** [comment]
Yes, sure. I'm trying to figure out how to DM  on discord

73 Huntington Road
YO31 8RL
07947 002 581
01904 673675

**Comment by ThosG** [comment]
> On 20 Apr 2019, at 18:01, Andy Miller <notifications@github.com> wrote:
> 
> I would need to see the underlying .md files though to know what is causing the issue. Can you zip it up and send it to me? perhaps via DM on Discord chat?
> 
I’ve put a zipped copy of the pages folder here on Dropbox:
https://www.dropbox.com/sh/6andgy280p7t2p3/AAAqUbTIEwWyqrkY4q0wvmgVa?dl=0 <https://www.dropbox.com/sh/6andgy280p7t2p3/AAAqUbTIEwWyqrkY4q0wvmgVa?dl=0>

Sorry, I couldn’t figure how to do it via DM on Discord.

Many thanks for your prompt support, by the way.

Thomas

**Comment by rhukster** [comment]
ok, pretty clear the problem is quite widespread in this theme.. you have pages that have:

```yaml
content:
   items:
       '@page.descendants': '/cantatas'
       '@page.descendants': '/largerworks'
       '@page.descendants': '/concertedvoices'
       '@page.descendants': '/arrangements'
```

This is not legal because `@page.descendants` is an array key, and you can't have duplicate array keys.

All of these pages shoudl have structure like this:

```yaml
content:
   items:
      -  '@page.descendants': '/cantatas'
      -  '@page.descendants': '/largerworks'
      -  '@page.descendants': '/concertedvoices'
      -  '@page.descendants': '/arrangements'
```

**Comment by ThosG** [comment]
Ah yes, well found Andy.

Luckily BBEdit enjoys doing edits on multiple files so this might be fairly quick to fix (ho ho, said the Fates, who does he think he’s kidding?)

Is there any tool out there to check the Yamliness of a folder of pages?  That could be pretty useful.

Anyway, many thanks Andy. I have a little monthly payment set up for Grav, and I have just had a good return on investment :)

Thomas

**Comment by rhukster** [comment]
That YAML Linter idea was a good one...  I have added a new CLI command, plus a new section to the Admin plugin's `Tools -> Reports` section that will show you any linting errors in Grav configuration or pages.

```bash
$ bin/grav yamllinter                                                                                                                [17:38:59]

Yaml Linter
===========

User Configuration
------------------


 [ERROR] YAML Linting issues found...


/user/config/test.yaml - Duplicate key "foo" detected at line 2 (near "foo: baz").

Pages Frontmatter
-----------------


 [OK] No YAML Linting issues with pages

```

This is now in `develop` branch and will be available in Grav 1.6.7

**Comment by ThosG** [comment]
👍

73 Huntington Road
YO31 8RL
07947 002 581
01904 673675
